Wire-iOS/Sources/UserInterface/Conversation/InputBar/MentionsHandler.swift

```diff
@@ -18,6 +18,12 @@
 
 import Foundation
 
+extension String {
+    var wholeRange: NSRange {
+        return NSRange(location: 0, length: endIndex.utf16Offset(in: self))
```

Comment:
Are we sure we want to do it like this? The approach with passing the range and the string is probably the most encoding-agnostic way, and lets Apple take care of the underlying implementation.

```suggestion
        return NSRange(startIndex ..< endIndex, in: self)
```

Reply:
There is more info on [SE-241](https://github.com/apple/swift-evolution/blob/master/proposals/0241-string-index-explicit-encoding-offset.md), but the main idea in this case is that we need to make sure we use UTF16 indexes when constructing  `NSRange`, because this is the encoding that `NSString` is using. Your proposed change would be fine if we would be creating a `Range` and using only Swift APIs, but in this case we are passing it to `NSRegularExpression` which is an Objective-C class working with `NSString`.

Reply:
I misread the proposed change a bit and it seems indeed that it's a cleaner API that basically does the same under the hood - https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSRange.swift#L257-L261